app/src/main/java/ch/epfl/sweng/SDP/game/drawing/DrawingOffline.java

```diff
@@ -0,0 +1,128 @@
+package ch.epfl.sweng.SDP.game.drawing;
+
+import android.content.Context;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.RequiresApi;
+import android.support.annotation.VisibleForTesting;
+import android.util.Log;
+import android.view.View;
+import android.widget.TextView;
+
+import ch.epfl.sweng.SDP.R;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.ValueEventListener;
+
+public class DrawingOffline extends DrawingActivity implements SensorEventListener {
+
+    private int speed;
+    private SensorManager sensorManager;
+
+    protected final ValueEventListener listenerTimer = new ValueEventListener() {
+        @Override
+        public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+            Integer value = dataSnapshot.getValue(Integer.class);
+
+            if (value != null) {
+                ((TextView) findViewById(R.id.timeRemaining)).setText(String.valueOf(value));
+            }
+        }
+
+        @Override
+        public void onCancelled(@NonNull DatabaseError databaseError) {
+            // Does nothing for the moment
+        }
+    };
+
```

Comment:
We do not really need a timer for offline mode, do we ? If not this whole block could be removed